parsl/executors/ipp.py

```diff
@@ -243,9 +243,9 @@ def scale_out(self, blocks=1):
                                         "Attempts to provision nodes via provider has failed"))
                 self.engines.extend([block])
                 r.extend([block])
-        else:
-            logger.error("No execution provider available")
-            r = None
+            else:
```

Comment:
You'd think python would complain about this!

Reply:
it's valid syntax - `for` loops can have `else` clauses in python. Which I learned because of this PR.

Reply:
TIL.